kvserver: cleanup replica checksum computation func

pav-kv opened the pull request:
This commit makes the replica checksum helper more convenient for tests, and simplifies the tests using it.

Follows up on https://github.com/cockroachdb/cockroach/pull/89813#discussion_r997404648

Epic: None
Release note: None

pav-kv committed:
kvserver: cleanup replica checksum computation func

This commit makes the replica checksum helper more convenient for tests, and
simplifies the tests using it.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/90313)
<!-- Reviewable:end -->

erikgrinaker commented on pkg/kv/kvserver/replica_consistency.go:
```diff
@@ -475,21 +475,23 @@ func (*Replica) computeChecksumDone(rc *replicaChecksum, result *replicaHash) {
 }
 
-// replicaSHA512 computes the SHA512 hash of the replica data at the given
+// calcReplicaDigest computes the SHA512 hash of the replica data at the given
```
```suggestion
// calcReplicaDigest computes the SHA512 hash and MVCC stats of the replica data at the given
```

pavelkalinnikov replied:
Done.

erikgrinaker commented on pkg/kv/kvserver/helpers_test.go:
```diff
@@ -551,14 +552,11 @@ func WatchForDisappearingReplicas(t testing.TB, store *Store) {
-func ChecksumRange(
+// CalcReplicaDigest computes a digest of the replicated data for the given
+// range. For testing only.
+func CalcReplicaDigest(
```
nit: mild preference for getting rid of this and exporting `calcReplicaDigest`. Test callers can just set up a `noopLimiter` in the global scope and reuse that.

pavelkalinnikov replied:
Agree. I wonder if it's a commonly used pattern, and it would make sense to have some `Unlim` limiter right in the main package.

pavelkalinnikov replied:
Doesn't appear so. This type is not widely used.

pavelkalinnikov replied:
Done. The `quotapool` limiter already has code to handle infinite `rate`, so I think it makes sense to add a package constant, similarly to how it's done in Go's limiter: https://pkg.go.dev/golang.org/x/time/rate#pkg-constants. Will split this PR in 2 commits.

erikgrinaker reviewed (approved):
Thanks!

pav-kv committed:
util/quotapool: introduce infinite limit helper

This commit adds quotapool.Inf() which can be used in tests to create an
unlimited RateLimiter.

Release note: None

pavelkalinnikov commented:
bors r=erikgrinaker
